CMakeLists.txt

```diff
@@ -179,6 +179,45 @@ if (PYBIND11_TEST)
   add_subdirectory(tests)
 endif()
 
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
+if(CMAKE_VERSION VERSION_GREATER 3.0.1)
+  # Build an interface library target:
+  #   Though any project using pybind11 will need to include python headers and
+  #   define c++11/14-ness, these are not included as interface includes or compile
+  #   defs because (1) that's really the domain of the upstream project to decide,
+  #   (2) we can get away with it since installation (headers) don't care about
+  #   python version or C++ standard, and (3) this will make the easily target relocatable.
+  add_library(pb11 INTERFACE)
+  set_target_properties(pb11 PROPERTIES EXPORT_NAME "pybind11")
+  target_include_directories(pb11 INTERFACE $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+  if(APPLE)
+      target_link_libraries(pb11 INTERFACE "-undefined dynamic_lookup")
```

Comment:
That can't be done exactly here because we don't want paths to system libraries embedded in the ConfigTarget*.cmake file. If it's going to be done, Python will need to be redetected in the pybind11Config.cmake and appended there. See full comment for more on this possibility.